Paper: MDAKits: A Framework for FAIR-Compliant Molecular Simulation Analysis

ianmkenney opened the pull request:
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.


*Editor*: <!--editor-->@deniederhut<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@yuanqing-wang<!--end-reviewers-list-->

orbeckst committed:
whitepaper as basis for scipy proceedings

- copied over whitepaper v0.1
- translated from latex to reST
- minor adjustments
- title and abstract adjusted to scipy submission

orbeckst committed:
add MDAKit examples

- snapshot of frontend
- add link
- list example packages
- add forgotten references

orbeckst committed:
add missing references for example kits

ianmkenney committed:
Fixed named references

lilyminium committed:
minor updates

IAlibay committed:
Merge pull request #42 from MDAnalysis/update-manuscript

Minor updates to tenses

deniederhut commented:
@scoobies assign @deniederhut as editor

scoobies commented:
Assigned! @deniederhut is now the editor

deniederhut commented:
@scoobies build status

scoobies commented:
{"MDAnalysis-2023_mdakits"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmpo3etk0ax'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: ian_kenney\n*** Warning: /tmp/tmpo3etk0ax/publisher/../output/ian_kenney/paper_stats.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/MDAnalysis-2023_mdakits.pdf", "build_status"=>"success", "build_timestamp"=>"03/06 20:34"}, "status"=>"success"}}

deniederhut commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cpc.2014.01.018 is OK
- 10.6084/m9.figshare.791567 is OK
- 10.1186/s13059-022-02625-x is OK
- 10.25080/Majora-629e541a-00e is OK
- 10.1002/jcc.21787 is OK
- 10.25080/majora-1b6fd038-005 is OK
- 10.25080/shinma-7f4c6e7-00a is OK
- 10.25080/Majora-7ddc1dd1-013 is OK
- 10.15497/RDA00068 is OK
- 10.1038/sdata.2016.18 is OK
- 10.1021/acs.jcim.0c01000 is OK
- 10.5281/zenodo.4570556 is OK
- 10.1063/5.0019088 is OK
- 10.1186/s13321-021-00548-6 is OK
- 10.1016/j.bbadva.2021.100010 is OK
- 10.1021/acs.jctc.1c00447 is OK
- 10.5281/zenodo.4545322 is OK
- 10.1371/journal.pcbi.1004415 is OK
- 10.1021/ct400341p is OK
- 10.1016/j.softx.2015.06.001 is OK
- 10.1038/s41592-019-0506-8 is OK
- 10.5281/zenodo.4685131 is OK
- 10.1016/j.bpj.2014.05.037 is OK
- 10.5281/ZENODO.4774216 is OK
- 10.5281/zenodo.3555620 is OK
- 10.33011/livecoms.3.1.1473 is OK
- 10.1103/PhysRevE.65.031910 is OK
- 10.1142/S0218195995000064 is OK
- 10.1080/00268978300102721 is OK
- 10.1073/pnas.1311599110 is OK
- 10.1080/00268979400100764 is OK
- 10.1063/5.0008316 is OK
- 10.1021/jp027149c is OK
- 10.1021/jp027149c is OK
- 10.1021/acs.jctc.8b00933 is OK
- 10.1021/la9408681 is OK
- 10.1002/prot.340090204 is OK
- 10.1063/1.4922445 is OK
- 10.1007/BF03018603 is OK
- 10.1016/j.cplett.2011.04.066 is OK
- 10.1021/ja211328g is OK
- 10.1002/jcc.21777 is OK
- 10.1063/1.3569857 is OK
- 10.1039/C3FD00145H is OK
- 10.1039/C9CP01532A is OK
- 10.1371/journal.pcbi.1004568 is OK
- 10.1002/jcc.21439 is OK
- 10.1002/(SICI)1097-0134(19990901)36:4<419::AID-PROT5>3.0.CO;2-U is OK
- 10.1529/biophysj.104.052449 is OK
- 10.1016/j.str.2007.12.011 is OK
- 10.1051/sfn/201112010 is OK
- 10.21105/joss.00877 is OK
- 10.1021/jp037128y is OK
- 10.3233/APC200023 is OK
- 10.1109/MCSE.2018.2881905 is OK
- 10.1016/j.commatsci.2015.09.013 is OK
- 10.1016/j.commatsci.2020.110086 is OK
- 10.1038/s41597-020-00638-4 is OK
- 10.1038/s42254-022-00463-1 is OK
- 10.1016/j.cpc.2013.09.018 is OK
- 10.1016/j.jmb.2013.10.024 is OK

MISSING DOIs

- 10.1080/07391102.2000.10506570 may be a valid DOI for title: HELANAL: A Program to Characterize Helix Geometry in Proteins
- 10.1016/s0263-7855(97)00009-x may be a valid DOI for title: HOLE: A program for the analysis of the pore dimensions of ion channel structural models

INVALID DOIs

- https://doi.org/10.1016/j.jmb.2009.09.009 is INVALID because of 'https://doi.org/' prefix
```

deniederhut commented:
@scoobies add @blaiszik to reviewers

scoobies commented:
@blaiszik added to the reviewers list!

deniederhut commented:
@scoobies code of conduct

scoobies commented:
Our CoC: https://www.scipy2023.scipy.org/code-of-conduct

scoobies commented:
Reports of abusive or harassing behavior may be reported to scipy@enthought.com

deniederhut commented:
@scoobies add @yuanqing-wang to reviewers

scoobies commented:
@yuanqing-wang added to the reviewers list!

deniederhut commented:
Hey @blaiszik and @yuanqing-wang! Friendly reminder that your initial review is due June 23

yuanqing-wang commented on papers/ian_kenney/ian_kenney.rst:
```diff
@@ -0,0 +1,494 @@
+
+   The reproducibility and transparency of scientific findings are widely recognized as crucial for promoting scientific progress.
+   However, when it comes to scientific software, researchers face many barriers and few incentives to ensure that their software is open to the community, thoroughly tested, and easily accessible.
+   To address this issue, the "MDAKits" framework has been developed, which simplifies the process of creating toolkits for the MDAnalysis simulation analysis package (https://www.mdanalysis.org/) that follow the basic principles of FAIR (findability, accessibility, interoperability, and reusability).
```
the quotation mark here should really be backticks

lilyminium replied:
Thanks, we have updated the manuscript accordingly.

yuanqing-wang commented on papers/ian_kenney/ian_kenney.rst:
```diff
@@ -0,0 +1,494 @@
+
+One solution is to consolidate scientific code around a small number of large, central packages. MDAnalysis :cite:`michaud-agrawal_mdanalysis_2011, gowers_mdanalysis_2016` is a widely-used open-source Python library for molecular simulation data. With over 16 years of development by more than 160 developers, MDAnalysis has refined its code base to offer a mature, robust, flexible API that offers  ...
+
+Until recently, MDAnalysis encouraged users to contribute their code back into the library to make it available to others. Notable examples of this include the waterdynamics :cite:`araya-secchi_characterization_2014` and ENCORE :cite:`tiberti_encore_2015` analysis modules. This approach, also successfully taken by packages such as cpptraj :cite:`roe_ptraj_2013` and the GROMACS tools :cite:`abraha ...
```
meaning that recently they stopped encouraging contributions?

lilyminium replied:
Apologies for the confusion -- we mean that cpptraj and GROMACS have both encouraged contributions to a central package, which has worked well for them, and didn't mean to imply that they have recently stopped. We have updated the manuscript with this clarification.

lilyminium replied:
In addition, we would like to clarify that MDAnalysis still welcomes contributions to the main library -- we have just expanded our focus to also encouraging code to be released as individual packages. The manuscript has been updated to reflect this as well.

yuanqing-wang commented on papers/ian_kenney/ian_kenney.rst:
```diff
@@ -0,0 +1,494 @@
+- The necessity of keeping the API stable between major releases precludes quick releases of breaking changes. In general, a mature package such as MDAnalysis has a slow release cycle, so new features and bug fixes can take months to become available in new releases.
+- As MDAnalysis implicitly agrees to maintain any code that has been added, a certain level of understanding and expertise is required from the maintainers. If the core developer team lacks expertise in a specific discipline or subdiscipline, adding new code in these areas introduces a substantial maintenance burden should the original code contributors not be available to help with maintenance.  ...
+- Introducing new package dependencies incurs software stack maintenance costs for many users who may not require this additional code.
+- Code contributors lose complete ownership of their code.
```
I think it's difficult to define "ownership"

lilyminium replied:
Thanks. Here we mean the ability to independently assert changes on the code base. We have updated the text to instead say “control” to reflect this.

yuanqing-wang commented on papers/ian_kenney/ian_kenney.rst:
```diff
@@ -0,0 +1,494 @@
+
+- Ensuring that the code follows best practices often requires long review periods and strict code-style adherence, thus slowing down the availability of the new code in a released version of the package.
+- The necessity of keeping the API stable between major releases precludes quick releases of breaking changes. In general, a mature package such as MDAnalysis has a slow release cycle, so new features and bug fixes can take months to become available in new releases.
+- As MDAnalysis implicitly agrees to maintain any code that has been added, a certain level of understanding and expertise is required from the maintainers. If the core developer team lacks expertise in a specific discipline or subdiscipline, adding new code in these areas introduces a substantial maintenance burden should the original code contributors not be available to help with maintenance.  ...
```
"implicitly agrees to maintain" builds on top of the basis that they were PR'ed into the main branch, right?

lilyminium replied:
Thanks for this question. To clarify here, we mean that the MDAnalysis project agrees to maintain any code that it releases. We have slightly updated the text to reflect this.

yuanqing-wang commented on papers/ian_kenney/ian_kenney.rst:
```diff
@@ -0,0 +1,494 @@
+..    .. _`fig:metadatapropkatraj`:
+
+
+After a pull request is opened, the MDAnalysis developers review the contents of the submission based on the following criteria:
```
off the point, but are these process automated?

lilyminium replied:
Most of these checks are partially automated through our CI pipelines. This is something we are continually improving as we obtain more community contributions and explore the diversity of our downstream ecosystem. However, we believe that there are elements of these checks that cannot, and should not, be automated -- for example, the evaluation of community guideline breaches. In many ways this outlines the necessity of having direct human intervention through dedicated members of the community willing to continually review and support the registry, rather than relying solely on automatic discovery tooling such as Github’s dependency graph, or analyzing PyPi / conda-forge data.

yuanqing-wang reviewed (commented):
This paper is very well structured and written. It addresses a critical problem in open source community that are well within the scope of the conference using innovative methods.

orbeckst committed:
references fixes

- fixed DOIs (thank you @scoobies)
- resolved an abbreviated journal name

deniederhut commented:
@scoobies remove @blaiszik from reviewers

scoobies commented:
@blaiszik removed from the reviewers list!

deniederhut commented:
@scoobies mark pending comment

Commit:
remove "file" fields from references

Commit:
change misc to software or online tags

Commit:
add howpublished

Commit:
add notes

Commit:
fix up yaml

Commit:
tweak formatting

Commit:
update yaml

Commit:
rm line numbers

Commit:
try replacing yaml with picture

Commit:
update caption

orbeckst committed:
author list update

- rearranged author list, following consensus from meeting on 2023-06-28
- added equal-contributor for the first four authors (as discussed)
- added ORCID identifiers (although they do not seem to show up in generated PDF)

orbeckst committed:
add reference for OpenMM

orbeckst committed:
have two intro sentences in Introduction

Just moved two sentences up and reworded next sentence to avoid hanging "this".

orbeckst committed:
changed writing style from whitepaper to proceedings

- replaced most future tense with present tense
- removed any references to whitepaper/planning (we have implemented what we said in
  the original whitepaper)

deniederhut commented:
Hi @yuanqing-wang! How are you feeling about the updates from the authors?

IAlibay committed:
Update ian_kenney.rst

IAlibay committed:
Merge pull request #43 from MDAnalysis/update_authors

Update authors count

IAlibay committed:
update with Oliver's wording

IAlibay committed:
Merge pull request #44 from MDAnalysis/reviewer_comments

Update with Oliver's wording on community growth.

IAlibay committed:
ialibay instead of IAlibay

IAlibay committed:
Small text updates

orbeckst committed:
Merge pull request #45 from MDAnalysis/final-read

Small text updates

fiona-naughton committed:
spelling, add footnote

fiona-naughton committed:
update FAIR reference year

fiona-naughton committed:
restoring removed backticks

Commit:
changes

Commit:
more changes

lilyminium committed:
Merge pull request #46 from MDAnalysis/final_fixes

lilyminium committed:
Merge pull request #47 from MDAnalysis/lily-final-read

deniederhut commented:
@lilyminium is this paper ready for inclusion in the proceedings?

deniederhut commented:
@yuanqing-wang is this paper ready for inclusion in the proceedings?

deniederhut commented:
@scoobies mark ready

orbeckst committed:
updated funder acknowledgement to preferred form

CZI asked all grant holders to use grant DOIs in the acknowledgements so
we updated to the funder preferred line.

orbeckst commented:
Our funder, CZI, asked all grant holders to use grant DOIs in the acknowledgements so we updated to the funder's requested text. No other changes were made to the manuscript and hence none of scientific content has changed.

deniederhut commented:
Thanks for the heads up

deniederhut reviewed (approved):
Thanks @yuanqing-wang for helping us get this paper ready for publication! And thanks @ianmkenney for joining us at SciPy this year!

![1bf9e96e58db30a4c7112b0e0397ac6d-2372965617](https://github.com/scipy-conference/scipy_proceedings/assets/6868396/c301c489-8d03-4732-844a-f6f40b66b249)
